core/src/main/java/org/apache/iceberg/avro/ValueReaders.java

```diff
@@ -381,6 +550,45 @@ public BigDecimal read(Decoder decoder, Object ignored) throws IOException {
       byte[] bytes = bytesReader.read(decoder, null);
       return new BigDecimal(new BigInteger(bytes), scale);
     }
+
+    @Override
+    public void skip(Decoder decoder) throws IOException {
+      bytesReader.skip(decoder);
+    }
+  }
+
+  private static class RequiredOptionReader implements ValueReader<Object> {
```

Comment:
Why do we need this next to the UnionReader?

Reply:
This is actually unused so we could remove it. The purpose is to be able to replace the union reader with one that checks that the value is non-null for cases where the file has an optional field but the expected schema requires it.

With Iceberg's schema evolution rules, we should never have that case, which is why I didn't end up using this (it was complicated and of little value). But I included the class just in case we want it in the future.

It would be good to hear what you think. Should we keep or remove it?

Reply:
Since you've approved this and I don't see any other required changes, I'm going to remove this to unblock getting this commit in.

Reply:
I would remove it, the PR looks good, so feel free to merge it once the tests are green 👍